### PR TITLE
Remove names from translatable strings

### PR DESCRIPTION
**Pull request by hebasto**
This PR removes the following tokens from translatable strings:
- command line option names
- RPC names
- debug log file name

All of them are untranslatable by their nature.

The Translation Strings Policy updated accordingly.

This was done while reviewing #20401. Good to get it just after 0.21 branch off :)

### CODE REVIEW

**Comment by decryp2kanon** [comment]
ACK c20e3db 

Good idea :+1:

File: src/wallet/wallet.cpp [hunk 1]
```diff
@@ -4041,7 +4041,7 @@ std::shared_ptr<CWallet> CWallet::Create(interfaces::Chain& chain, const std::st
             }
 
             if (rescan_height != block_height) {
-                error = _("Prune: last wallet synchronisation goes beyond pruned data. You need to -reindex (download the whole blockchain again in case of pruned node)");
+                error = strprintf(_("Prune: last wallet synchronisation goes beyond pruned data. You need to %s (download the whole block chain again in case of pruned node)."), "-reindex");
```
Comment (decryp2kanon) [inline src/wallet/wallet.cpp:4044]: nit: blockchain vs block_chain

Comment (hebasto) [reply src/wallet/wallet.cpp:4044]: https://github.com/bitcoinops/bitcoinops.github.io/blob/master/STYLE.md

**Comment by MarcoFalke** [comment]
If we want to avoid this in the future, would a linter be appropriate?

**Comment by laanwj** [comment]
I understand the idea but… Doesn't this make it much harder for translators to understand what is being translated? A lot of context information goes missing by replacing these with placeholders.

File: src/init.cpp [hunk 2]
```diff
@@ -1416,16 +1420,16 @@ bool AppInitMain(const util::Ref& context, NodeContext& node, interfaces::BlockA
     }
     strSubVersion = FormatSubVersion(CLIENT_NAME, CLIENT_VERSION, uacomments);
     if (strSubVersion.size() > MAX_SUBVERSION_LENGTH) {
-        return InitError(strprintf(_("Total length of network version string (%i) exceeds maximum length (%i). Reduce the number or size of uacomments."),
-            strSubVersion.size(), MAX_SUBVERSION_LENGTH));
+        return InitError(strprintf(_("Total length of network version string (%i) exceeds maximum length (%i). Reduce the number or size of %s."),
```
Comment (luke-jr) [inline src/init.cpp:1423]:
```suggestion
        return InitError(strprintf(_("Total length of network version string (%i) exceeds maximum length (%i). Reduce the number or size of %s options."),
```

File: doc/translation_strings_policy.md [hunk 3]
```diff
@@ -30,6 +30,15 @@ This includes messages passed to the GUI through the UI interface through `InitM
 General recommendations
 ------------------------
 
+### Do not translate names
+
+The following tokens must not be parts of the strings are to be translated:
+- command line option names
+- RPC names
```
Comment (luke-jr) [inline doc/translation_strings_policy.md:37]:
```suggestion
- RPC method and parameter names
```

File: src/init.cpp [hunk 4]
```diff
@@ -1011,18 +1011,19 @@ bool AppInitParameterInteraction(const ArgsManager& args)
...
     // if using block pruning, then disallow txindex
     if (args.GetArg("-prune", 0)) {
-        if (args.GetBoolArg("-txindex", DEFAULT_TXINDEX))
-            return InitError(_("Prune mode is incompatible with -txindex."));
+        if (args.GetBoolArg("-txindex", DEFAULT_TXINDEX)) {
+            return InitError(strprintf(_("Prune mode is incompatible with %s."), "-txindex"));
+        }
         if (!g_enabled_filter_types.empty()) {
-            return InitError(_("Prune mode is incompatible with -blockfilterindex."));
+            return InitError(strprintf(_("Prune mode is incompatible with %s."), "-blockfilterindex"));
```
Comment (luke-jr) [inline src/init.cpp:1026]: Should we combine these (like `ResolveErrMsg`)?

File: src/init.cpp [hunk 5]
```diff
@@ -1446,12 +1450,13 @@ bool AppInitMain(const util::Ref& context, NodeContext& node, interfaces::BlockA
...
         if (!Lookup(proxyArg, proxyAddr, 9050, fNameLookup)) {
-            return InitError(strprintf(_("Invalid -proxy address or hostname: '%s'"), proxyArg));
+            return InitError(strprintf(_("Invalid %s address or hostname: '%s'"), "-proxy", proxyArg));
         }
 
         proxyType addrProxy = proxyType(proxyAddr, proxyRandomize);
-        if (!addrProxy.IsValid())
-            return InitError(strprintf(_("Invalid -proxy address or hostname: '%s'"), proxyArg));
+        if (!addrProxy.IsValid()) {
+            return InitError(strprintf(_("Invalid %s address or hostname: '%s'"), "-proxy", proxyArg));
```
Comment (luke-jr) [inline src/init.cpp:1458]: Another candidate for combining

File: src/init.cpp [hunk 6]
```diff
@@ -1470,11 +1475,12 @@ bool AppInitMain(const util::Ref& context, NodeContext& node, interfaces::BlockA
...
             CService onionProxy;
             if (!Lookup(onionArg, onionProxy, 9050, fNameLookup)) {
-                return InitError(strprintf(_("Invalid -onion address or hostname: '%s'"), onionArg));
+                return InitError(strprintf(_("Invalid %s address or hostname: '%s'"), "-onion", onionArg));
             }
             proxyType addrOnion = proxyType(onionProxy, proxyRandomize);
-            if (!addrOnion.IsValid())
-                return InitError(strprintf(_("Invalid -onion address or hostname: '%s'"), onionArg));
+            if (!addrOnion.IsValid()) {
+                return InitError(strprintf(_("Invalid %s address or hostname: '%s'"), "-onion", onionArg));
```
Comment (luke-jr) [inline src/init.cpp:1482]: combine

File: src/wallet/wallet.cpp [hunk 7]
```diff
@@ -2961,7 +2961,7 @@ bool CWallet::CreateTransactionInternal(
                 nFeeNeeded = GetMinimumFee(*this, nBytes, coin_control, &feeCalc);
                 if (feeCalc.reason == FeeReason::FALLBACK && !m_allow_fallback_fee) {
                     // eventually allow a fallback fee
-                    error = _("Fee estimation failed. Fallbackfee is disabled. Wait a few blocks or enable -fallbackfee.");
+                    error = strprintf(_("Fee estimation failed. %s is disabled. Wait a few blocks or enable %s."), "-fallbackfee", "-fallbackfee");
```
Comment (luke-jr) [inline src/wallet/wallet.cpp:2964]: First one should probably be translated...

File: src/wallet/wallet.cpp [hunk 8]
```diff
@@ -3935,7 +3935,7 @@ std::shared_ptr<CWallet> CWallet::Create(interfaces::Chain& chain, const std::st
     if (gArgs.IsArgSet("-discardfee")) {
         CAmount nFeePerK = 0;
         if (!ParseMoney(gArgs.GetArg("-discardfee", ""), nFeePerK)) {
-            error = strprintf(_("Invalid amount for -discardfee=<amount>: '%s'"), gArgs.GetArg("-discardfee", ""));
+            error = strprintf(_("Invalid amount for %s=<amount>: '%s'"), "-discardfee", gArgs.GetArg("-discardfee", ""));
```
Comment (luke-jr) [inline src/wallet/wallet.cpp:3938]: This message seems reusable

File: src/wallet/wallet.cpp (hunk 1 above)
Comment (luke-jr) [reply src/wallet/wallet.cpp:4044]: This isn't optech

**Review by luke-jr — CHANGES_REQUESTED** [review]
Concept ACK

> Doesn't this make it much harder for translators to understand what is being translated? A lot of context information goes missing by replacing these with placeholders.

Qt lets you specify an additional string seen only by translators for context. Does Transifex support that? Can we add it to our utils too?

**Comment by hebasto** [comment]
@laanwj 
> I understand the idea but… Doesn't this make it much harder for translators to understand what is being translated? A lot of context information goes missing by replacing these with placeholders.

I understand your concerns. Yes, translators will lose some part of context with this approach.
OTOH, for https://github.com/bitcoin/bitcoin/blob/7d8a10a6f4d61df1f4afb138e379149565afa02d/src/init.cpp#L1200

just spotted the following translation:

DeepinScreenshot_select-area_20210317011131

> (the **most common** kind of question that I get on transifex is already "I don't understand the context of this message, please explain". It might also result in less specific and relevant translations.)

Does answering those questions resolve translators' lack of context?

**Comment by laanwj** [comment]
> Does answering those questions resolve translators' lack of context?

Not really—in the sense that it's not a scalable approach, and also many will not ask and just assume something.
If there is some other way to provide context that is shown in the translator (like the translation comment) I guess that would work though.

**Comment by hebasto** [comment]
> If there is some other way to provide context that is shown in the translator (like the translation comment) I guess that would work though.

From https://docs.transifex.com/localization-tips-workflows/the-complete-guide-to-adding-context-to-strings
> DEVELOPER NOTES... These types of notes are added to source files by developers. Not all file formats support developer notes though. The file types which support developer notes are the following: Android XML, Apple PLIST, Apple Strings, JSON (Chrome), Structured JSON, PO, Mozilla DTD, Windows Resx, Resw and Resjson, Xliff, xlsx (as context) and YAML.

Qt files are not mentioned, unfortunately.

**Comment by hebasto** [comment]
Closing for now. Leaving up for grabs.

**Pull request closed without merging**